ssl/record/methods/dtls_meth.c

```diff
@@ -365,6 +365,44 @@ static int dtls_retrieve_rlayer_buffered_record(OSSL_RECORD_LAYER *rl,
     return 0;
 }
 
+/* rfc9147 section 4.2.3 */
+int dtls_crypt_sequence_number(unsigned char *seq, size_t seq_len,
+                               EVP_CIPHER_CTX *ctx, unsigned char *rec_data)
+{
+    unsigned char mask[16];
+    int outlen, inlen;
+    unsigned char *iv, *in;
+    const char *name = EVP_CIPHER_get0_name(EVP_CIPHER_CTX_get0_cipher(ctx));
+    size_t i;
+
+    memset(mask, 0, sizeof(mask));
+
+    if (strncmp(name, "AES", 3) == 0) {
+        iv = NULL;
+        in = rec_data;
+        inlen = 16;
+    } else if (strncmp(name, "Cha", 3) == 0) {
+        iv = rec_data;
+        in = rec_data + 4;
+        inlen = 12;
+    } else {
+        return 0;
+    }
```

Comment:
@fwh-dc can you please fix/comment on this?

Reply:
Yes sorry for the wait:
Good comment.

Regarding NULL cipher: 
Is it valid for a (D)TLS 1.3 to have a NULL cipher? If not I'll remove the fetching support. Otherwise your suggestion makes sense to implement.

Regarding cipher name comparisons:
I'd really like to be able to do something like (kind like of `find / -name "foo*"`):
```
EVP_cipher_is_a(c, "AES-256-*");
EVP_cipher_is_a(c, "AES-*");
EVP_cipher_is_a(c, "*-256-*");
```

But it does not look like that is supported. I think I'll go for your suggestion `OPENSSL_strncasecmp()` which is less tedious than maintaining a complete list of ciphers.

Reply:
> Regarding NULL cipher:
> Is it valid for a (D)TLS 1.3 to have a NULL cipher? If not I'll remove the fetching support. Otherwise your suggestion makes sense to implement.

As NULL (authenticated-only) ciphersuites were added for TLS-1.3, IMO they should be supported for DTLS-1.3 as well. I do not see a reason why they should not work with it.

Reply:
Well then I will add it as a no-op thanks!